smart-hadoop-support/smart-hadoop-client/src/main/java/org/smartdata/hdfs/client/SmartDFSClient.java

```diff
@@ -129,13 +133,18 @@ public DFSInputStream open(String src, int buffersize,
       boolean verifyChecksum)
       throws IOException, UnresolvedLinkException {
     FileState fileState;
+    src = checkS3ForXattr(src);
```

Comment:
I think we need to move this behind `fileState = new S3FileState(src)`, otherwise the src may be wrong.